### PR TITLE
chore(sdk-core): refactor BaseCoin from bitgo to sdk-core.

### PR DESCRIPTION
**Pull request by ducnguyen-bitgo**
- Moved baseCoin to sdk-core.
- Moved enterprise and enterprises to sdk-core.
- Moved internal and keycard to sdk-core.
- Moved keychains to sdk-core.
- Moved markets to sdk-core.
- Moved pendingApproval and pendingApprovals to sdk-core.
- Moved affirmation and affirmations to sdk-core.
- Moved settlement and settlements to sdk-core.
- Moved tradingAccount to sdk-core.
- Moved tradingPartner, and tradingPartners to sdk-core.
- Moved blsUtils, mpcUtils, opengpgUtils, promise-utils to sdk-core.
- Moved tssUtils, util to sdk-core.
- Moved wallet, and wallets to sdk-core.
- Moved webhooks to sdk-core.
- Moved bip32util, bitcoin, ecdh, and tss to sdk-core.

Ticket: BG-48167

### CODE REVIEW

File: modules/sdk-core/src/bitgo/utils/abstractUtxoCoinUtil.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import * as utxolib from '@bitgo/utxo-lib';
+import ScriptType2Of3 = utxolib.bitgo.outputScripts.ScriptType2Of3;
+import { isChainCode, scriptTypeForChain } from '@bitgo/utxo-lib/dist/src/bitgo';
```
Comment (bitgoAaron) [inline modules/sdk-core/src/bitgo/utils/abstractUtxoCoinUtil.ts:3]: let's avoid importing from `/dist` folders. Can we instead make sure utxo-lib exports these types correctly?

File: modules/sdk-core/package.json [hunk 2]
```diff
@@ -37,15 +37,21 @@
...
     "@stablelib/hex": "^1.0.0",
+    "big.js": "^3.1.3",
     "bignumber.js": "^9.0.0",
     "bip32": "^2.0.6",
     "bitcoinjs-lib": "npm:@bitgo/bitcoinjs-lib@6.1.0-rc.3",
+    "bitcoinjs-message": "^2.0.0",
     "bs58": "^4.0.1",
     "create-hmac": "^1.1.7",
+    "debug": "^3.1.0",
+    "libsodium-wrappers-sumo": "^0.7.9",
```
Comment (bitgoAaron) [inline modules/sdk-core/package.json:50]: same here. Remove this from bitgo's package.json

**Comment by bitgoAaron** [comment]
looks like all of the files under `sdk-core/src/account-lib/mpc` were manually copied instead of `git mv`, thus losing the history. Can you redo these as well? We need to keep track of as much as possible

**Review by bitgoAaron — CHANGES_REQUESTED** [review]
see comments

File: modules/sdk-core/package.json (hunk 2 above)
Comment (ducnguyen-bitgo) [reply modules/sdk-core/package.json:50]: I think `libsodium-wrappers-sumo` is from account-lib's package.json. I'm removing it now.

File: modules/sdk-core/src/bitgo/utils/abstractUtxoCoinUtil.ts (hunk 1 above)
Comment (ducnguyen-bitgo) [reply modules/sdk-core/src/bitgo/utils/abstractUtxoCoinUtil.ts:3]: Those functions are exported under `utxo-lib/src/bitgo` so I guess we could just do `import { bitgo } from '@bitgo/utxo-lib'` and then destructure those functions from `bitgo`?

File: modules/sdk-core/src/api/bip32path.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
+/**
```
Comment (bitgoAaron) [inline modules/sdk-core/src/api/bip32path.ts:1]: the file should be `git mv`'d, not copy/pasted.

**Review by bitgoAaron — CHANGES_REQUESTED** [review]
2 more copy/pasted files need to be git mv'd to preserve history.

File: modules/sdk-core/src/bitgo/utils/util.ts [hunk 4]
```diff
@@ -0,0 +1,199 @@
+/**
```
[the remark this answers is not in the document]
Comment (ducnguyen-bitgo) [reply modules/sdk-core/src/bitgo/utils/util.ts:1]: @bitgoAaron The reason we didn't git-mv it was to keep it backward compatible for bitgo-express until the next release of sdk-core.

**Review by bitgoAaron — APPROVED** [review]

**Pull request merged**